examples/generate.sh

```diff
@@ -217,31 +229,8 @@ echo "Generated ${CONTROLPLANE_GENERATED_FILE}"
 kustomize build "${SOURCE_DIR}/machinedeployment" | envsubst >> "${MACHINEDEPLOYMENT_GENERATED_FILE}"
 echo "Generated ${MACHINEDEPLOYMENT_GENERATED_FILE}"
 
-cp ${SOURCE_DIR}/addons.yaml "${ADDONS_GENERATED_FILE}"
-echo "Generated ${ADDONS_GENERATED_FILE}"
-
-# Generate Cluster API provider components file.
-CAPI_BRANCH=${CAPI_BRANCH:-"stable"}
-if [[ ${CAPI_BRANCH} == "stable" ]]; then
-  curl -L https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.2.4/cluster-api-components.yaml > "${COMPONENTS_CLUSTER_API_GENERATED_FILE}"
-  echo "Downloaded ${COMPONENTS_CLUSTER_API_GENERATED_FILE} from cluster-api stable branch - v0.2.4"
-else
-  kustomize build "github.com/kubernetes-sigs/cluster-api/config/default/?ref=${CAPI_BRANCH}" > "${COMPONENTS_CLUSTER_API_GENERATED_FILE}"
-  echo "Generated ${COMPONENTS_CLUSTER_API_GENERATED_FILE} from cluster-api - ${CAPI_BRANCH}"
-fi
-
-# Generate Kubeadm Bootstrap Provider components file.
-CABPK_BRANCH=${CABPK_BRANCH:-"stable"}
-if [[ ${CABPK_BRANCH} == "stable" ]]; then
-  curl -L https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/releases/download/v0.1.2/bootstrap-components.yaml > "${COMPONENTS_KUBEADM_GENERATED_FILE}"
-  echo "Downloaded ${COMPONENTS_KUBEADM_GENERATED_FILE} from cluster-api-bootstrap-provider-kubeadm stable branch - v0.1.2"
-else
-  kustomize build "github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/config/default/?ref=${CABPK_BRANCH}" > "${COMPONENTS_KUBEADM_GENERATED_FILE}"
-  echo "Generated ${COMPONENTS_KUBEADM_GENERATED_FILE} from cluster-api-bootstrap-provider-kubeadm - ${CABPK_BRANCH}"
-fi
-
 # Generate OpenStack Infrastructure Provider components file.
-kustomize build "${SOURCE_DIR}/../config/default" | envsubst > "${COMPONENTS_OPENSTACK_GENERATED_FILE}"
+kustomize build "${SOURCE_DIR}/../config" | envsubst > "${COMPONENTS_OPENSTACK_GENERATED_FILE}"
 echo "Generated ${COMPONENTS_OPENSTACK_GENERATED_FILE}"
```

Comment:
KUBERNETES_VERSION in line 28 should be over 1.16, because minimum Kubernetes version supporting CRDv1 is v1.16.
https://cluster-api.sigs.k8s.io/developer/providers/v1alpha2-to-v1alpha3.html#upgrade-to-crd-v1

Reply:
generate.sh does not add cloud-config secret to cluster.yaml. I am not sure when it happened. we can fix by follow up patch.

Reply:
The next PR would replace the whole example generation stuff with the templating done by clusterctl. But it's good to know :) Fixed the Kubernetes version

Reply:
Thanks for fixing the Kubernetes version.